src/app/static/src/app/store/surveyAndProgram/actions.ts

```diff
@@ -0,0 +1,38 @@
+import {ActionContext, ActionPayload, ActionTree} from 'vuex';
+import {RootState} from "../../main";
+import {SurveyAndProgramDataState} from "./surveyAndProgram";
+import {GeoJSON} from "geojson";
+import {api} from "../../apiService";
+
+export type SurveyAndProgramActionTypes = "SurveyLoaded" | "SurveyError"
+
+export interface SurveyAndProgramPayload extends ActionPayload {
+    type: SurveyAndProgramActionTypes
+}
+
+export interface SurveyLoaded extends SurveyAndProgramPayload {
+    geoJson: string
+}
+
+export interface SurveyError extends SurveyAndProgramPayload {
+    error: string
+}
+
+export interface SurveyAndProgramActions {
+    uploadSurvey: (store: ActionContext<SurveyAndProgramDataState, RootState>, file: File) => void
+}
+
+export const actions: ActionTree<SurveyAndProgramDataState, RootState> & SurveyAndProgramActions = {
+
+    uploadSurvey({commit}, file) {
+        let formData = new FormData();
+        formData.append('file', file);
+        api.post<GeoJSON>("/survey/", formData)
```

Comment:
Ha, this confused me for a moment - GeoJSON is the type of payload we're expecting in the response, not what we're expect to post in the request as it looked like to me.. Renaming this method might make that more obvious but also more verbose: postAndReturn<GeoJSON> or something?